zebra-chain/src/block/header.rs

```diff
@@ -67,17 +68,40 @@ pub struct Header {
     pub solution: Solution,
 }
 
+/// TODO: Inline this error into zebra_consensus::error::BlockError.
```

Comment:
My intuition is that we might not want to inline this, and instead take the first error message defined here, "invalid time in block header" and make that the error message on the variant in `BlockError`. Then we make `BlockTimeError` `#[source]` for `BlockError::Time` and have it only print "must be less than 2 hours in the future {0:?}" as its error message. This way we can let the reporter split the msg into multiple lines if it wants to which will make our error reports nicer imo.

The hint bit should _really_ be handled by the Generic Member Access RFC and things like this always make me sad that this isn't a feature we have yet in rust :(((, for now though I think shoving it in the error message like you did here is the best option.

Reply:
```suggestion
/// TODO: Use this error as the source for zebra_consensus::error::BlockError::Time,
/// and make `BlockError::Time` add additional context.
```

Reply:
I moved the details to the follow-up ticket #1021.